pkg/helpers/imageregistry/imageregistry.go

```diff
@@ -0,0 +1,81 @@
+package imageregistry
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/open-cluster-management/multicloud-operators-foundation/pkg/apis/imageregistry/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func AddToScheme(s *runtime.Scheme) error {
+	if err := clusterv1.Install(s); err != nil {
+		return err
+	}
+	if err := v1alpha1.AddToScheme(s); err != nil {
+		return err
+	}
+	return nil
+}
+
+type ImageRegistryClient struct {
+	client client.Client
+}
+
+func NewImageRegistry(client client.Client) *ImageRegistryClient {
+	return &ImageRegistryClient{
+		client: client,
+	}
+}
+
+// Get returns the registry address and pullSecret.
+// found is true if the imageRegistry in the label open-cluster-management.io/image-registry of cluster exists.
+func (c *ImageRegistryClient) Get(clusterName string) (registryAddress, pullSecretNamespace, pullSecretName string, found bool, err error) {
```

Comment:
why not merge Get and OverrideImage together?

Reply:
refactor to 4 methods:
GetPullSecret(clusterName string) (pullSecretNamespace, pullSecretName string, err error)
GetRegistry(clusterName string) (registry string, err error) 
OverrideImage(clusterName, imageName string) (newImageName string, err error) 
GetImageRegistry(clusterName string) (*v1alpha1.ManagedClusterImageRegistry, error)